tests/magma/runtime/test_runtime.py

```diff
@@ -12,7 +13,10 @@ class TestRuntime(unittest.TestCase):
     def test_runtime_creation(self):
```

Comment:
Unfortunately, I don't have time right now to go through all the unit tests...

But please add a high level docstring to each test saying what the test is about in one sentence. 

In addition, I find it useful to document with comments throughout the unit test what is happening and why we do what we do to communicate the intent.
The goal is for somebody to still understand in half a year why we have written a test the way we did. This can often get quite convoluted when testing detailed features.

So spell out that we are doing XYZ because we want to test a specific aspect of an API.

See for example how I tested the AbstractProcess, Ports or Compiler.